scipy/signal/_peak_finding.py

```diff
@@ -943,7 +869,7 @@ def find_peaks(x, height=None, threshold=None, distance=None,
     >>> plt.show()
     """
     # _argmaxima1d expects array of dtype 'float64'
-    x = np.asarray(x, dtype=np.float64)
+    x = np.asarray(x, order="C", dtype=np.float64)
```

Comment:
Why enforce C-ordering here?

Reply:
As mentioned in the commit message:
> Ensure that x is C-contiguous. This way it doesn't need to get  converted twice for prominence and width calculation. Later this might  prove useful if the input of `_argmaxima1d` is declared as C-contiguous.

But I can revert this and introduce it once `_argmaxima1d` actually requires C-contiguous arrays. Your call.

Reply:
I'd leave it out for now.